test/test_bitfield.cpp

```diff
@@ -262,3 +262,67 @@ TORRENT_TEST(test_resize_down)
 		TEST_EQUAL(test1.count(), i / 2);
 	}
 }
+
+TORRENT_TEST(find_first_set_empty)
+{
+	bitfield test1(0);
+	TEST_EQUAL(test1.find_first_set(), -1);
+}
+
+TORRENT_TEST(find_first_set_small)
+{
+	bitfield test1(10, false);
+	TEST_EQUAL(test1.find_first_set(), -1);
+}
+
+TORRENT_TEST(find_first_set_large)
+{
+	bitfield test1(100, false);
+	TEST_EQUAL(test1.find_first_set(), -1);
+}
+
+TORRENT_TEST(find_first_set_early)
+{
+	bitfield test1(100, false);
+	test1.set_bit(4);
+	TEST_EQUAL(test1.find_first_set(), 4);
+}
+
+TORRENT_TEST(find_first_set_late)
+{
+	bitfield test1(100, false);
+	test1.set_bit(98);
+	TEST_EQUAL(test1.find_first_set(), 98);
+}
+
+TORRENT_TEST(find_last_clear_empty)
+{
+	bitfield test1(0);
+	TEST_EQUAL(test1.find_last_clear(), -1);
+}
+
+TORRENT_TEST(find_last_clear_small)
+{
+	bitfield test1(10, true);
+	TEST_EQUAL(test1.find_last_clear(), -1);
+}
+
+TORRENT_TEST(find_last_clear_large)
+{
+	bitfield test1(100, true);
+	TEST_EQUAL(test1.find_last_clear(), -1);
+}
+
+TORRENT_TEST(find_last_clear_early)
+{
+	bitfield test1(100, true);
+	test1.clear_bit(4);
+	TEST_EQUAL(test1.find_last_clear(), 4);
+}
+
+TORRENT_TEST(find_last_clear_late)
```

Comment:
maybe it would make sense to have a case where both bit 4 and 98 are cleared, to make sure it returns the index to the last one